### PR TITLE
[SMC-22] Improve exception logging

### PR DESCRIPTION
**Pull request by dinvlad**
This PR adds stacktraces to logged exceptions.

https://broadworkbench.atlassian.net/browse/SMC-22

### CODE REVIEW

**Review by zbedo — APPROVED** [review]

File: src/utils/google_cloud/google_cloud_compute.py [hunk 1]
```diff
@@ -59,8 +59,8 @@ def delete_everything(self) -> None:
                 .list(project=self.gcp_project, region=constants.SERVER_REGION)
                 .execute()["items"]
             )
-        except Exception as e:
-            logger.error(f"Error getting subnets: {e}")
+        except:
+            logger.exception("Error getting subnets:")
```
Comment (TomConner) [inline src/utils/google_cloud/google_cloud_compute.py:63]:
```suggestion
            logger.exception("Error getting subnets")
```

File: src/utils/studies_functions.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ async def email(inviter: str, recipient: str, invitation_message: str, study_tit
         return response.status_code  # type: ignore
 
     except HTTPError as e:
-        logger.error(f"Email failed to send: {e}")
+        logger.error(f"Email failed to send: {e}", exc_info=True)
```
Comment (TomConner) [inline src/utils/studies_functions.py:73]: Was `{e}` left in intentionally in this case?

File: src/utils/google_cloud/google_cloud_compute.py (hunk 1 above)
Comment (TomConner) [reply src/utils/google_cloud/google_cloud_compute.py:63]: I suppose if the stack trace is logged, then you want the colon and should reject this suggestion. 

If you want to remove the colon then it ought to be removed in the other error messages in this PR too.

**Review by TomConner — APPROVED** [review]
Good error message improvement. I'm assuming the stack traces are not sent to the browser in production.

File: src/utils/studies_functions.py (hunk 2 above)
Comment (dinvlad) [reply src/utils/studies_functions.py:73]: Ah yep! Since it's `logger.error()` still

**Pull request merged**